### ISSUE TITLE
Scoping Issue with nested loops

### ISSUE THREAD

**Issue by chbloemer** [neuland/jade4j#191]
```
for a in ["A","B"]
    - var b = a
    for c in ["C","D"]
        - b = b + c;
    p=b
```
Should result in:
```
<p>ACD</p>
<p>BCD</p>
```

But the result is:
```
<p>AB</p>
<p>AB</p>
```

**Comment by chbloemer** [comment]
The first one actually works in jade4j. But this one does not. If I put an if in front of the loop and reassign a variable it doesn't work.
```
- var value = ""
if true
  - value = "Hallo"
p=value

- var x = "A"
for a in ["B","C"]
  - x = x + a
p=x
```
Should result in:
```
<p>Hallo</p>
<p>ABC</p>
```

But the result is:
```
<p>Hallo</p>
<p>A</p>
```

**Comment by chbloemer** [comment]
Fixed in 1.3.1
